.github/actions/start-local-server/action.yml

```diff
@@ -22,8 +22,7 @@ runs:
     - name: Wait for boot run to be ready
       id: wait-for-ready
       run: |
-        timeout 60 bash -c 'until echo > /dev/tcp/localhost/8080; do sleep 1; done'
-        resultStatus=$(echo > /dev/tcp/localhost/8080; echo $?)
+        resultStatus=$(timeout 60 bash -c 'until echo > /dev/tcp/localhost/8080; do sleep 1; done' && echo $?)
```

Comment:
I'm trying to translate into English what this does. You're saying wait up to a minute, checking every second, for localhost:8080 to spit something out. If that works, echo the exit code, which should be zero if it worked. And the thing that changed is that now we're getting the exit status of the first command and not another `echo > /dev/tcp/localhost/8080`.

I'm still confused.

Reply:
I'll take my own medicine and add some comment. Let me try to explain...

Since `timeout` is a top-level bash command in the `run:`, when it times out it SIGTERMs which terminates the github action.

By putting the `timeout` in a subprocess, it just exits the subprocess. As a result, `resultStatus` is either empty, because the subprocess terminated, or it is 0 if we got an answer back from the echo.